examples/ColorGradientPlots.py

```diff
@@ -0,0 +1,139 @@
+        self.sample_idx = next_idx
+
+        sample_phase_pos = sample_phases / (self.period*self.rate)
+        sample_phase_pos %= 1.0
+        if self.neg_period is None:
+            return sample_phase_pos**4
+        sample_phase_neg = sample_phases / (self.neg_period*self.rate)
+        sample_phase_neg %= 1.0
+        return sample_phase_pos**4 - sample_phase_neg**4
+
+class MainWindow(pg.GraphicsLayoutWidget):
+    """ example application main window """
+    def __init__(self):
+        super().__init__()
+        self.setWindowTitle('pyqtgraph example: gradient plots')
+        self.resize(800,800)
+        self.show()
+        
+        layout = self # we are using a GraphicsLayoutWidget as main window for convenience
+        cm = pg.colormap.get('CET-L17')
+        cm.reverse()
+        pen0 = cm.getPen( span=(0.0,1.0), width=5 )
+        curve0 = pg.PlotDataItem(pen=pen0 ) # vertical linear color map
+
+        cm = pg.colormap.get('CET-D1')
+        cm.setMappingMode('diverging')
+        brush = cm.getBrush( span=(-1., 1.), orientation='vertical' ) 
+        curve1 = pg.PlotDataItem(pen='w', brush=brush, fillLevel=0.0 ) # horizonal mirrored color map
+        
+        cm = pg.colormap.get('CET-L17')
+        cm.setMappingMode('mirror')
+        pen2 = cm.getPen( span=(400.0,600.0), width=5, orientation='horizontal' )
+        curve2 = pg.PlotDataItem(pen=pen2 ) # horizonal repeating color map
+
+        cm = pg.colormap.get('CET-C2')
+        cm.setMappingMode('repeat')
+        pen3 = cm.getPen( span=(100, 200), width=5, orientation='horizontal' )
+        curve3 = pg.PlotDataItem(pen=pen3 ) # vertical diverging fill
+
+        curves = (curve0, curve1, curve2, curve3)
+
+        length = int( 3.0 * 200. ) # length of display in samples
+        self.top_plot = None
+        for idx, curve in enumerate( curves ):
+            plot = layout.addPlot(row=idx+1, col=0)
+            if self.top_plot is None:
+                self.top_plot = plot
+            else:
+                plot.setXLink( self.top_plot )
+            plot.addItem( curve )
+            plot.setXRange( 0, length )
+            if idx != 1: plot.setYRange(  0. , 1. )
+            else       : plot.setYRange( -1. , 1. ) # last plot include positive/negative data
+
+        self.traces = (
+            {'crv': curve0, 'buf': np.zeros( length ), 'ptr':0, 'ds': DataSource( signal_period=0.55 ) },
+            {'crv': curve1, 'buf': np.zeros( length ), 'ptr':0, 'ds': DataSource( signal_period=0.61, negative_period=0.55 ) },
+            {'crv': curve2, 'buf': np.zeros( length ), 'ptr':0, 'ds': DataSource( signal_period=0.65 ) },
+            {'crv': curve3, 'buf': np.zeros( length ), 'ptr':0, 'ds': DataSource( signal_period=0.52 ) },
+        )
+        self.timer = QtCore.QTimer(timerType=QtCore.Qt.PreciseTimer)
+        self.timer.timeout.connect(self.update)
+        timestamp = time.time()
```

Comment:
Probably should use `time.perf_counter()` for float type, or `time.perf_counter_ns()`  for int type (in ns).  Also we have [ptime.py](https://github.com/pyqtgraph/pyqtgraph/blob/master/pyqtgraph/ptime.py) but I'm not sure it's actually relevant any more or not...

Reply:
(already updated)